File: tags/NHI1-0.9/theLink/example/php/mulserver.php
<?php
#+
#§  \file       theLink/example/php/mulserver.php
#§  \brief      \$Id$
#§  
#§  (C) 2010 - NHI - #1 - Project - Group
#§  
#§  \version    \$Rev$
#§  \author     EMail: aotto1968 at users.berlios.de
#§  \attention  this software has GPL permissions to copy
#§              please contact AUTHORS for additional information
#§

class MulServer extends MqS implements iServerSetup, iFactory {
  public function Factory() {
    return new MulServer();
  }
  public function ServerSetup() {
    $this->ServiceCreate('MMUL', array(&$this, 'MMUL'));
  }
  public function MMUL() {
    $this->SendSTART();
    $this->SendD($this->ReadD() * $this->ReadD());
    $this->SendRETURN();
  }
}
$srv = new MulServer();
try {
  $srv->LinkCreate($argv);
  $srv->ProcessEvent(MqS::WAIT_FOREVER);
} catch (Exception $ex) {
  $srv->ErrorSet($ex);
}
$srv->Exit();

?>

File: tags/NHI1-0.11/theLink/example/ruby/mulserver.rb
#+
#§  \file       theLink/example/ruby/mulserver.rb
#§  \brief      \$Id: LbMain 26 2009-12-03 11:48:43Z aotto1968 $
#§  
#§  (C) 2010 - NHI - #1 - Project - Group
#§  
#§  \version    \$Rev: 26 $
#§  \author     EMail: aotto1968 at users.berlios.de
#§  \attention  this software has GPL permissions to copy
#§              please contact AUTHORS for additional information
#§

require "rubymsgque"

class MulServer < MqS
  def initialize(tmpl = nil)
    super()
    ConfigSetServerSetup(method(:ServerSetup))
  end
  def MMUL
    SendSTART()
    SendD(ReadD() * ReadD())
    SendRETURN()
  end
  def ServerSetup()
    ServiceCreate("MMUL",method(:MMUL))
  end
end

srv = FactoryNew("mulserver", MulServer)
begin
  srv.LinkCreate($0,ARGV)
  srv.ProcessEvent(MqS::WAIT_FOREVER)
rescue Exception => ex
  srv.ErrorSet(ex)
ensure
  srv.Exit()
end

